### PR TITLE
More Either tests

### PR DESCRIPTION
**Pull request by a-ivanov**
Hi everyone!

My first PR. Improving Either test coverage.

### CODE REVIEW

File: arrow-libs/core/arrow-core/src/test/kotlin/arrow/core/EitherTest.kt [hunk 1]
```diff
@@ -162,14 +229,78 @@ class EitherTest : UnitSpec() {
...
+          Left(a).replicate(n) == Left(a)
+      }
+    }
+
+    "traverse should return list of Right when Right and empty list when Left" {
+      forAll(
+        Gen.int(),
+        Gen.int(),
+        Gen.int()
+      ) { a: Int, b: Int, c: Int ->
```
Comment (a-ivanov) [inline arrow-libs/core/arrow-core/src/test/kotlin/arrow/core/EitherTest.kt:287]: Can test with `Gen.list`, but I thought that showing list of `Right` is more explicit.

File: arrow-libs/core/arrow-core/src/test/kotlin/arrow/core/EitherTest.kt [hunk 2]
```diff
@@ -323,3 +454,12 @@ private suspend fun <A> throwException(
   a: A
 ): Either<Throwable, Any> =
   throw RuntimeException("An Exception is thrown while handling the result of the supplied function.")
+
+private fun forAllSmallInt(fn: PropertyContext.(a: Int) -> Boolean) =
```
Comment (raulraja) [inline arrow-libs/core/arrow-core/src/test/kotlin/arrow/core/EitherTest.kt:458]: I believe we have some other utilities like these somewhere. Perhaps these functions should be moved there and made public or internal.

**Review by raulraja — APPROVED** [review]
Thank you so much @a-ivanov and welcome to Arrow! 🙏

**Comment by rachelcarmena** [comment]
Welcome to Arrow @a-ivanov !! :tada:

**Comment by rachelcarmena** [comment]
And thank you so much!! :raised_hands:

**Comment by a-ivanov** [comment]
Thanks everyone for the warm welcome! Hope not to disappoint in future)

**Pull request merged**